opa/opa.go

```diff
@@ -51,51 +62,47 @@ func (v *PolicyViolation) Write(w io.Writer) {
 }
 
 // NewClient OpaClient constructor
-func NewClient(logger *zap.Logger, host string) *Client {
-	client := &Client{
+func NewClient(logger *zap.Logger, host string) Client {
+	client := &client{
 		logger: logger,
 		Host:   host,
 	}
 	return client
 }
 
 // InitializePolicy initializes OPA policy if it does not already exist
-func (opa *Client) InitializePolicy(policy string) ClientError {
+func (opa *client) InitializePolicy(policy string) ClientError {
 	_ = opa.logger.Named("Initialize Policy")
 
 	exists, err := opa.policyExists(policy)
 	if err != nil {
-		return clientError{"error checking if policy exists", OpaClientErrorTypePolicyExists, err}
+		return NewClientError("error checking if policy exists", OpaClientErrorTypeGetPolicy, err)
 	}
 
 	if !exists {
-		// fetch violations from ES
-		violations := []PolicyViolation{}
-		err = opa.publishPolicy(policy, violations)
-		if err != nil {
-			return clientError{"error publishing policy", OpaClientErrorTypePublishPolicy, err}
-		}
+		// TODO: fetch rules from ES
+		return NewClientError("policy does not exist", OpaClientErrorTypePolicyNotFound, nil)
 	}
 
 	return nil
 }
 
 // EvaluatePolicy evalutes OPA policy agains provided input
-func (opa *Client) EvaluatePolicy(policy string, input string) (*EvaluatePolicyResult, error) {
+func (opa *client) EvaluatePolicy(policy string, input []byte) (*EvaluatePolicyResponse, error) {
 	log := opa.logger.Named("Evalute Policy")
 	request, err := json.Marshal(&EvalutePolicyRequest{Input: json.RawMessage(input)})
 	if err != nil {
-		log.Error("failed to encode OPA input", zap.Error(err), zap.String("input", input))
+		log.Error("failed to encode OPA input", zap.Error(err), zap.String("input", string(input)))
```

Comment:
nitpick: instead of logging errors and then returning those same errors in this package, I think we should just return them, and expect the consumer to log those errors out.  it's easier if all of the logging is done in one spot.